mlx/backend/common/cholesky.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include "mlx/allocator.h"
+#include "mlx/backend/common/copy.h"
+#include "mlx/linalg.h"
+#include "mlx/primitives.h"
+
+#ifdef ACCELERATE_NEW_LAPACK
+#include <Accelerate/Accelerate.h>
+#else
+#include <lapack.h>
+#endif
+
+namespace mlx::core {
+
+void cholesky_impl(const array& a, array& T, bool upper) {
```

Comment:
Nit: `T` is kind of a bad name for a variable such as this. `out` or `factor` or something along these lines is probably better.